### DISCUSSION TITLE
Issue with 0.10.0 S3 Authentication

### DISCUSSION THREAD

**Discussion by JRocki** [duckdb/duckdb#10696] [Q&A] [answered]
Hello,

I'm looking at this [page](https://duckdb.org/docs/extensions/httpfs/s3api) and trying to implement the new method for authentication by doing this:

```
con = duckdb.connect()
con.query(
    f"""
    CREATE SECRET secret (
    TYPE S3,
    PROVIDER CREDENTIAL_CHAIN);
    """
)
df = con.query("SELECT * FROM read_parquet('s3://my-bucket/file.parquet');")

```
This is failing, resulting in an error:
`duckdb.duckdb.IOException: IO Error: Connection error for HTTP HEAD to 'https://my-bucket./file.parquet'`

Anyone have any ideas? I can still connect using the legacy method, but I have to add a bunch of extra code to handle all the different chains myself.

**Comment by samansmink** [comment]
Hey @JRocki, could you give a bit more detail on where you credentials are stored? Are they in environment variables? or in you `~/.aws/credentials` file?

also does the `call load_aws_credentials()` function find your credentials properly?

**Reply by fj-sanchez** [reply]
I have the same issue... This is what I get from the `call load_aws_credentials()` execution

```
D call load_aws_credentials();
100% ▕████████████████████████████████████████████████████████████▏ 
┌──────────────────────┬──────────────────────────┬──────────────────────┬───────────────┐
│ loaded_access_key_id │ loaded_secret_access_key │ loaded_session_token │ loaded_region │
│       varchar        │         varchar          │       varchar        │    varchar    │
├──────────────────────┼──────────────────────────┼──────────────────────┼───────────────┤
│                      │                          │                      │ eu-west-1     │
└──────────────────────┴──────────────────────────┴──────────────────────┴───────────────┘
```

I use `credential_process = aws-okta cred-process XXXX` in my aws config.

**Reply by samansmink** [reply]
@fj-sanchez it appears that there was no key loaded there. does you current setup allow you to authenticate with the aws cli? im not familiar with aws-okta tbh

**Reply by fj-sanchez** [reply]
Yes, I can run from the CLI `aws s3 ls`, however I have to set the `AWS_PROFILE` env var (whcih I have also set before launching duckdb).

**Reply by samansmink** [reply]
ah, could you maybe try:

```sql
CREATE SECRET (
    TYPE S3, 
    PROVIDER CREDENTIAL_CHAIN, 
    PROFILE '<name_of_your_profile>'
)
```

**Reply by fj-sanchez** [reply]
Same result... is there any way to see any debug logs or similar? I can also confirm that if I execute in an environment with all the AWS_XX env vars set it can load the credentials and read the file (I guess that's equivalent to config mode).

I do it like this `aws-okta exec my-profile -- duckdb`.

**Reply by JRocki** [reply]
I haven't been able to find a work around (solution below doesn't work for me), but it might be useful to you.

**Comment by JRocki** [comment]
Thanks for replying! I authenticate using `aws sso login` and my credentials are stored in `~/.aws`. I have the `config` file which looks like this (replaced with filler text):
```
[default]
sso_start_url = https://url.here
sso_region = us-west-2
sso_account_id = 123
sso_role_name = XXX
region = us-west-1
output = json
```

`call load_aws_credentials()` successfully loads my credentials.

**Comment by JRocki** [comment]
If I run:
```
D INSTALL httpfs;
D LOAD httpfs;
D CREATE SECRET (
      TYPE S3,
      PROVIDER CREDENTIAL_CHAIN
  );
```
I get:
```
┌─────────┐
│ Success │
│ boolean │
├─────────┤
│ true    │
└─────────┘
```

**Comment by ned2** [comment, marked as answer]
hey @JRocki, my team ran into what looks like this problem. it looks like the region isn't getting added into the auto-generated S3 endpoint URI (even when this is specified in the REGION parameter of the `CREATE SECRET` command). we found we could workaround this by explicitly configuring the ENDPOINT using the S3 URI for our region:

```
CREATE SECRET (
    TYPE S3, 
    PROVIDER CREDENTIAL_CHAIN, 
    PROFILE '<name_of_your_profile>',
    ENDPOINT 's3.<your-region>.amazonaws.com'
)
```

@samansmink  possibly a bug?

**Reply by JRocki** [reply]
Thanks @ned2! I wasn't able to get that to work for me but I will keep looking into it to make sure it's not on my end. Glad you found a work-around.

**Reply by JRocki** [reply]
Oh so if I remove `PROFILE` it does work! Maybe just because I only use default. I will mark this as an answer for now.
